templatefitter/fit_model/parameter_handler.py

```diff
@@ -929,7 +961,8 @@ def used_by(
         template_parameter: TemplateParameter,
         template_serial_number: int,
     ) -> None:
-        template_parameter.base_model_parameter = self
+        if not template_parameter.base_model_parameter_is(self):
```

Comment:
`base_model_parameter` can be initialized in the Constructor of `TemplateParameter` now, this is also the way it's done in `FitModel`. 
However, to preserve the option of setting it externally via the public `used_by` I need to check if has already been set to the parameter I want to set it to (otherwise I will run into the assert in line 907). If it is `None` setting it should work, if it is something else I will trigger the assert.